csharp/src/Directory.Build.props

```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <LangVersion>10.0</LangVersion>
+    <LangVersion>11.0</LangVersion>
```

Comment:
Is this necessary for the change?

I can see that we may need to have at least one project targeting .NET 7, but I'd prefer not to make that a general requirement, sticking to LTS SDK versions if at all possible.

I think it would be reasonable for CI to install .NET 7 (as you've done above) to make sure that we really solve the AOT  issues, but I'd want everything other than that project to only require .NET 6, if at all possible.

Does that sound like a reasonable proposal that should work out the kinks on, or do you think it's unreasonable for folks to stay on the .NET 6 SDK?

(It would *definitely* make things simpler internally if we could stick to C# 10 features, at least for the main package.)

Reply:
I used list pattern matching in some tests. I can revert.

Reply:
Thank you - much appreciated :)

Reply:
(Although if that's only in the project that relies on .NET 7, it would be fine to specify LangVersion 11 in that specific project file.)

Reply:
The main Google.Protobuf csproj overrides this setting and forces C# 10 on itself. I think you already have the situation you desire.

By the way, I noticed that updating Google.Protobuf to use C# 11 introduced new compilation errors in Google.Protobuf. It doesn't like how the library uses ref structs. I didn't investigate.

Reply:
> I think you already have the situation you desire.

Well, not really. With this change it's "every project which doesn't specify anything gets C# 11, but Google.Protobuf specifies C# 10" whereas I'd prefer "every project which doesn't specify anything gets C# 10, but the AOT project gets C# 11". (This change would currently still be a problem internally where we'll be able to compile C# 10 but not C# 11 - we definitely want to be able to build Google.Protobuf.Test etc.)